client/src/components/profile/components/TimeLine.js

```diff
@@ -168,10 +144,38 @@ class Timeline extends Component {
   }
 }
 
+TimelineInner.propTypes = propTypes;
+
+function useIdToNameMap() {
+  const {
+    allChallengeNode: { edges }
+  } = useStaticQuery(graphql`
+    query challengeNodes {
+      allChallengeNode {
+        edges {
+          node {
+            fields {
+              slug
+            }
+            id
+            title
+          }
+        }
+      }
+    }
+  `);
+  const idToNameMap = new Map();
+  edges.forEach(({ node: { id, title, fields: { slug } } }) => {
+    idToNameMap.set(id, { challengeTitle: title, challengePath: slug });
+  });
+  return idToNameMap;
+}
+
+const Timeline = props => {
+  const idToNameMap = useIdToNameMap();
+  return <TimelineInner idToNameMap={idToNameMap} {...props} />;
```

Comment:
```suggestion
  return <TimelineInner idToNameMap={createIdToNameMap()} {...props} />;
```
See above.

Reply:
I've updated this to obey the convention.

Reply:
@ValeraS is it okay to pass a hook directly like this or is this a shortcut too far?

Reply:
not sure, but prefer don't do this with hooks.

Reply:
Otherwise, we may miss it on future refactors.

Reply:
I'm not 100% sure either, so let's play it safe.